### PR TITLE
Feat/core models/minossi

### PR DESCRIPTION
**Pull request by gordominossi**
Add some core models

### CODE REVIEW

File: core/tile-entity/init.lua [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    for address, componentType in component.list() do
+        if isKnownType(componentType) then
+            knownEntities[address] = component.proxy(address)
+        end
+    end
+end
+
+---Creates a new tileEntity object
+---@param address string
+---@param position coordinates
```
Comment (S4mpsa) [inline core/tile-entity/init.lua:34]: What is the format for `coordinates`? {x, y, z}, {x=number, y=number, z=number}? Something else?

File: core/tile-entity/inventory.model.lua [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+---Creates a new inventory object
+---@param address string
+---@param location coordinates
+---@return inventory
+function inventory.new(address, location)
+    ---@type inventory
+    local self = tileEntity.new(address, location)
+
+    local proxy = component.proxy(address)
+    self.name = proxy.getInventoryName()
```
Comment (S4mpsa) [inline core/tile-entity/inventory.model.lua]: This seems wrong, inventories by themselves do not usually have a component with an address, instead they are accessed using the address of a transposer and a side.

File: core/tile-entity/init.lua (hunk 1 above)
Comment (gordominossi) [reply core/tile-entity/init.lua:34]: The type is defined in the [types.lua](https://github.com/S4mpsa/NIDAS/blob/254e4bc796b5039318d5c09cf350cf24dbef32c7/core/tile-entity/types.lua) file

File: core/tile-entity/inventory.model.lua (hunk 2 above)
Comment (gordominossi) [reply core/tile-entity/inventory.model.lua]: 🤔

File: core/tile-entity/init.lua (hunk 1 above)
Comment (S4mpsa) [reply core/tile-entity/init.lua:34]: I see this is in `types.lua`

**Pull request merged**